test/unit/gateways/shift4_test.rb

```diff
@@ -127,6 +127,39 @@ def test_successful_purchase_with_stored_credential
     end.respond_with(successful_purchase_response)
   end
 
+  def test_successful_purchase_with_card_on_file_fields
```

Comment:
This is sufficient, I think, but it might be nice to add a test that has both sets of fields from `stored_credentials` and the GSF's and assert that the ones we put in the post body are what we are expecting.

Reply:
Thanks @naashton , the changes are applied.